### PR TITLE
Add Ripple Payment transaction and signing

### LINKED ISSUE #6: Add support for Ripple
Name: [Ripple](https://ripple.com)
HD Derivation Scheme: BIP44 / `144`
Symbol: `XRP`

Reference:
[Base58](https://github.com/ripple/rippled/blob/eed210bb67611aa36bdadfe14bfefb345d78a691/src/ripple/protocol/impl/tokens.cpp)

Checklist

- [x] Coin configuration
- - [x] coin type / decimals
- - [x] recommended tx block explorer ([bithomp](https://bithomp.com/explorer/6679490D06256510DF2F60D5A69F887CEEF6272EAB26C5EB03B7C16CF479A0C4))
- [x] Account / Address (https://developers.ripple.com/accounts.html)
- - [x] derived from bip39 phrases
- - [x] from public key / string
- - [x] validation
- - [x] C interfaces
- - [x] unit test
- [x] Transaction (https://developers.ripple.com/transaction-basics.html)
- - [x] serialization
- - [x] signing
- - [x] C interfaces
- - [x] unit test
- [ ] Integration
- - [x] Validate generated code in iOS projects
- - [ ] Validate generated code in Android projects
- - [x] Swift integration test
- - [ ] Java integration test

### PR DESCRIPTION
**Pull request by hewigovens**
## Description

Fix #6, we only support non negative / float XRP payment transaction from A to B.

Reference:
- https://developers.ripple.com/serialization.html#field-ids
- https://developers.ripple.com/transaction-common-fields.html
- https://github.com/trezor/trezor-core/tree/master/src/apps/ripple#transactions
- https://github.com/seelabs/rippled/blob/cecc0ad75849a1d50cc573188ad301ca65519a5b/src/ripple/protocol/impl/Serializer.cpp

## Testing instructions

All ripple UT passed

## Types of changes

- Add `Transaction`, `Signer` and `TWRippleSigner`

## Checklist

- [x] Prefix PR title with `[WIP]` if necessary.
- [x] Add tests to cover changes as needed.
- [x] Update documentation as needed.

### CODE REVIEW

File: src/Ripple/BinaryCoding.h [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+/// Encodes a 32-bit value into the provided buffer.
+static inline void encode32(uint32_t val, std::vector<uint8_t>& data) {
+    data.push_back(static_cast<uint8_t>((val >> 24)));
+    data.push_back(static_cast<uint8_t>((val >> 16) & 0xff));
+    data.push_back(static_cast<uint8_t>((val >> 8) & 0xff));
+    data.push_back(static_cast<uint8_t>(val));
+}
+
+/// Encodes a 64-bit value into the provided buffer.
+static inline void encode64(uint64_t val, std::vector<uint8_t>& data) {
```
Comment (alejandro-isaza) [inline src/Ripple/BinaryCoding.h]: We should consolidate this and the one under Bitcoin.

File: src/Ripple/Transaction.h [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+#pragma once
+
+#include "../proto/Ripple.pb.h"
+#include "../Data.h"
+#include "Address.h"
+
+namespace TW {
+namespace Ripple {
+
+enum FieldType: int {
```
Comment (alejandro-isaza) [inline src/Ripple/Transaction.h]: If this is only used in C++ let's use `class enum FieldType: int` and change enum cases to just the name like `int16`

File: src/Ripple/Transaction.h [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    FieldTypeInt16 = 1,
+    FieldTypeInt32 = 2,
+    FieldTypeAmount = 6,
+    FieldTypeVL = 7,
+    FieldTypeAccount = 8
+};
+
+enum TransactionType {
+    TransactionTypePayment = 0
```
Comment (alejandro-isaza) [inline src/Ripple/Transaction.h]: Same here

File: src/Ripple/BinaryCoding.h (hunk 1 above)
Comment (hewigovens) [reply src/Ripple/BinaryCoding.h]: Yes, the only difference is endian?

File: src/Ripple/Transaction.h (hunk 2 above)
Comment (hewigovens) [reply src/Ripple/Transaction.h]: done

File: src/Ripple/BinaryCoding.h (hunk 1 above)
Comment (alejandro-isaza) [reply src/Ripple/BinaryCoding.h]: Yes, let's have something like `encode64le` and `encode64be`

File: src/Ripple/BinaryCoding.h [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    data.push_back(static_cast<uint8_t>((val >> 32) & 0xff));
+    data.push_back(static_cast<uint8_t>((val >> 24) & 0xff));
+    data.push_back(static_cast<uint8_t>((val >> 16) & 0xff));
+    data.push_back(static_cast<uint8_t>((val >> 8) & 0xff));
+    data.push_back(static_cast<uint8_t>(val & 0xff));
+}
+
+/// Encodes a field type.
+static inline void encodeType(FieldType type, int key, std::vector<uint8_t>& data) {
+    int _type = int(type);
```
Comment (alejandro-isaza) [inline src/Ripple/BinaryCoding.h]:
```suggestion
    const auto typeValue = static_cast<int>(type);
```

**Pull request merged**